src/main/java/frc/robot/subsystems/Collector.java

```diff
@@ -5,34 +5,96 @@
 package frc.robot.subsystems;
 
 import com.ctre.phoenix6.controls.DutyCycleOut;
-import com.ctre.phoenix6.controls.PositionVoltage;
+import com.ctre.phoenix6.controls.VelocityVoltage;
+import com.ctre.phoenix6.sim.TalonFXSimState;
 
+import edu.wpi.first.hal.SimBoolean;
+import edu.wpi.first.hal.SimDevice;
+import edu.wpi.first.hal.SimDevice.Direction;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj.RobotBase;
+import edu.wpi.first.wpilibj.RobotController;
+import edu.wpi.first.wpilibj.simulation.LinearSystemSim;
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants.CollectorConstants;
+import frc.robot.Constants.RobotMap;
 import frc.thunder.hardware.ThunderBird;
+import frc.thunder.shuffleboard.LightningShuffleboard;
 
 
 public class Collector extends SubsystemBase {
 
     private ThunderBird motor;
-    public double power = 0;
+    public TalonFXSimState motorSim;
+    public SimDevice beamBreakSim;
+    public DigitalInput beamBreak;
```

Comment:
private most of your class vars, also reorganize all the variables here to be grouped nicely